test/test-lex.cpp

```diff
@@ -113,6 +113,149 @@ TEST_F(test_lex, lex_block_comments) {
+                              error_unopened_block_comment, comment_close,
+                              offsets_matcher(&input, 0, u8"*/"))));
+  }
+  {
+    error_collector v;
+    padded_string input(u8"foo(a, true * direct */);"_sv);
+    lexer l(&input, &v);  // identifier
+    EXPECT_EQ(l.peek().type, token_type::identifier);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::left_paren);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::identifier);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::comma);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::kw_true);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::star);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::identifier);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::right_paren);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::semicolon);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+    l.skip();
+    EXPECT_THAT(v.errors,
+                ElementsAre(ERROR_TYPE_FIELD(
+                    error_unopened_block_comment, comment_close,
+                    offsets_matcher(&input, strlen(u8"foo(a, true * direct "),
+                                    u8"*/"))));
+  }
+  {
+    error_collector v;
+    padded_string input(u8"**/"_sv);
+    lexer l(&input, &v);
+    EXPECT_EQ(l.peek().type, token_type::star);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+    EXPECT_THAT(v.errors, ElementsAre(ERROR_TYPE_FIELD(
+                              error_unopened_block_comment, comment_close,
+                              offsets_matcher(&input, 1, u8"*/"))));
+  }
+}
+
+TEST_F(test_lex, lex_unopened_block_comment2) {
+  {
+    error_collector v;
+    padded_string input(u8"3; */hello"_sv);
+    lexer l(&input, &v);  // number
+    EXPECT_EQ(l.peek().type, token_type::number);
+    l.skip();  // semicolon
+    EXPECT_EQ(l.peek().type, token_type::semicolon);
+    l.skip();  // identifier
+    EXPECT_EQ(l.peek().type, token_type::identifier);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+
+    EXPECT_THAT(v.errors,
+                ElementsAre(ERROR_TYPE_FIELD(
+                    error_unopened_block_comment, comment_close,
+                    offsets_matcher(&input, strlen(u8"3; "), u8"*/"))));
+  }
+}
+
+TEST_F(test_lex, lex_unopened_block_comment_with_regexp_literal) {
+  {
+    error_collector v;
+    padded_string input(u8"3; */ hello/"_sv);
+    lexer l(&input, &v);  // number
+    EXPECT_EQ(l.peek().type, token_type::number);
+    l.skip();  // semicolon
+    EXPECT_EQ(l.peek().type, token_type::semicolon);
+    l.skip();  // star
+    EXPECT_EQ(l.peek().type, token_type::star);
+    l.skip();  // slash
+    EXPECT_EQ(l.peek().type, token_type::slash);
+    l.skip();  // identifier
+    EXPECT_EQ(l.peek().type, token_type::identifier);
+    l.skip();  // slash
+    EXPECT_EQ(l.peek().type, token_type::slash);
+    l.skip();  // slash
```

Comment:
This comment is wrong. It should say 'end of file'.

Since the EXPECT_EQ right after this line already contains `end_of_file`, let's not write these comments.